Camera

teun2408 opened the pull request:
Should compute the viewing angle of a camera, still has some (major flaws):

- Does not use Binary search tree yet, just uses a lists that gets resorted after each operation.
- Does not work with all (viewing) angles yet. If it overflows 0/360 degrees weird stuff starts to happen.

teun2408 committed:
Added computation of the camera viewing angle. Still has some major flaws (does not use a bst and does not work with all camera angles)

dmbuzink commented on GeometricAlgorithmsGame/Assets/Scripts/Camera.cs:
```diff
@@ -52,10 +56,135 @@ void Update()
+
+        //Get all vertices, group them on their angle to the camera and sort them on their angle
+        var GroupedVertices = edges
+                .SelectMany(x => new List<Tuple<Vertex, Edge>>() { new Tuple<Vertex, Edge>(x.StartPoint, x), new Tuple<Vertex, Edge>(x.EndPoint, x) })
```
`new Tuple<Vertex, Edge>(x.EndPoint, x)` is wat overbodig aangezien `(v.EndPoint, x)` het zelfde doet. Van mij hoef je het niet aan te passen, maar dat je het weet voor in het vervolg ;)

dmbuzink commented on GeometricAlgorithmsGame/Assets/Scripts/Edge.cs:
```diff
@@ -10,5 +13,57 @@ public Edge(PolygonVertex startPoint, PolygonVertex endPoint)
+
+
+            //Computes the determinant between two vectors
+            double Det((double, double) a, (double, double) b)
```
Een method als het bereken van de `det` kan je ook prima ergens anders stoppen. Kan namelijk op andere vlekken ook wel handig zijn. (Again, van mij hoef je dit niet te verplaatsen ofso, maar meer in een algemene opmerking)

dmbuzink commented on GeometricAlgorithmsGame/Assets/Scripts/Edge.cs:
```diff
@@ -10,5 +13,57 @@ public Edge(PolygonVertex startPoint, PolygonVertex endPoint)
+
+            //Rather ugly, but seems to solve some of the annoying rounding issues
+            if (x > 16331239353195370 / 10000) x = 0;
+            if (y > 16331239353195370 / 10000) y = 0;
```
Misschien is wat uitleg handig waar deze getallen vandaan komen?

teun2408 replied:
Die zijn redelijk arbitrair, is eigenlijk gewoon de int.maxInt / 10000 omdat dat leek te werken na 1 test. Er moet echter wel iets in deze richting gebeuren. Ik ga het iets netter maken want dit is vrij lelijk inderdaad.

dmbuzink reviewed (approved):
Ik heb een paar comments geplaatst. Ziet er (snel overheen kijkend) wel goed uit.

teun2408 committed:
Moved the determinant to the geometric helper, added some explenation about arbitrary rounding variable.

teun2408 committed:
Made changes according to the discussion:
- Use the existing intersection method
- Use the vertices of the actual edges to add to the result if we add the start / end vertex.
- Order the events such that the first even is the minAngle event to 1PI. Then cycle around and get all events from -1PI to the min angle.

teun2408 committed:
Merge branch 'develop' into Camera

teun2408 committed:
draw cameraview

teun2408 committed:
Merge with develop

TarVK committed:
Made camera view continuously update

TarVK committed:
fixed typo

teun2408 committed:
Make the min/max angle their radial values consistent with the other fixing the bug where the cameras would only work for half of the angles.

teun2408 committed:
Fixed bug where edges are not correctly removed from the bst
